From 9db92b2b0c27420924bf95f4796303b83dfd2929 Mon Sep 17 00:00:00 2001
From: Federico Mena Quintero <federico@novell.com>
Date: Wed, 24 Mar 2010 14:25:10 -0600
Subject: [PATCH 1/1] bgo#594231 - Fix vertical orientation of VBoxes and other vertical widgets

We disable the "orientation" property for GtkVBox, GtkHBox, and all
the other concrete classes with a default orientation.  That property
will remain accessible for GtkBox and the other orientable base
classes.

Signed-off-by: Federico Mena Quintero <federico@novell.com>
---
 plugins/gtk+/gtk+.xml.in |   87 +++++++++++++++++++++++++++++++++++----------
 1 files changed, 67 insertions(+), 20 deletions(-)

diff --git a/plugins/gtk+/gtk+.xml.in b/plugins/gtk+/gtk+.xml.in
index d837035..724d488 100644
--- a/plugins/gtk+/gtk+.xml.in
+++ b/plugins/gtk+/gtk+.xml.in
@@ -1308,8 +1308,16 @@ embedded in another object</_tooltip>
 
     </glade-widget-class>
 
-    <glade-widget-class name="GtkHBox" generic-name="hbox" _title="Horizontal Box"/>
-    <glade-widget-class name="GtkVBox" generic-name="vbox" _title="Vertical Box"/>
+    <glade-widget-class name="GtkHBox" generic-name="hbox" _title="Horizontal Box">
+      <properties>
+	<property id="orientation" disabled="True"/>
+      </properties>
+    </glade-widget-class>
+    <glade-widget-class name="GtkVBox" generic-name="vbox" _title="Vertical Box">
+      <properties>
+	<property id="orientation" disabled="True"/>
+      </properties>
+    </glade-widget-class>
 
     <glade-widget-class name="GtkTable" generic-name="table" _title="Table" fixed="True">
       <post-create-function>glade_gtk_table_post_create</post-create-function>
@@ -1377,9 +1385,16 @@ embedded in another object</_tooltip>
       </packing-properties>
     </glade-widget-class>
 
-    <glade-widget-class name="GtkHPaned" generic-name="hpaned" _title="Horizontal Panes"/>
-
-    <glade-widget-class name="GtkVPaned" generic-name="vpaned" _title="Vertical Panes"/>
+    <glade-widget-class name="GtkHPaned" generic-name="hpaned" _title="Horizontal Panes">
+      <properties>
+	<property id="orientation" disabled="True"/>
+      </properties>
+    </glade-widget-class>
+    <glade-widget-class name="GtkVPaned" generic-name="vpaned" _title="Vertical Panes">
+      <properties>
+	<property id="orientation" disabled="True"/>
+      </properties>
+    </glade-widget-class>
   
     <glade-widget-class name="GtkNotebook" generic-name="notebook" _title="Notebook">
       <post-create-function>glade_gtk_notebook_post_create</post-create-function>
@@ -1465,9 +1480,16 @@ embedded in another object</_tooltip>
       </properties>
     </glade-widget-class>
     
-    <glade-widget-class name="GtkHScale" generic-name="hscale" _title="Horizontal Scale"/>
-    
-    <glade-widget-class name="GtkVScale" generic-name="vscale" _title="Vertical Scale"/>
+    <glade-widget-class name="GtkHScale" generic-name="hscale" _title="Horizontal Scale">
+      <properties>
+	<property id="orientation" disabled="True"/>
+      </properties>
+    </glade-widget-class>
+    <glade-widget-class name="GtkVScale" generic-name="vscale" _title="Vertical Scale">
+      <properties>
+	<property id="orientation" disabled="True"/>
+      </properties>
+    </glade-widget-class>
 
     <glade-widget-class name="GtkCalendar" generic-name="calendar" _title="Calendar">
       <properties>
@@ -1497,9 +1519,16 @@ embedded in another object</_tooltip>
 
     </glade-widget-class>
     
-    <glade-widget-class name="GtkHScrollbar" generic-name="hscrollbar" _title="Horizontal Scrollbar"/>
-
-    <glade-widget-class name="GtkVScrollbar" generic-name="vscrollbar" _title="Vertical Scrollbar"/>
+    <glade-widget-class name="GtkHScrollbar" generic-name="hscrollbar" _title="Horizontal Scrollbar">
+      <properties>
+	<property id="orientation" disabled="True"/>
+      </properties>
+    </glade-widget-class>
+    <glade-widget-class name="GtkVScrollbar" generic-name="vscrollbar" _title="Vertical Scrollbar">
+      <properties>
+	<property id="orientation" disabled="True"/>
+      </properties>
+    </glade-widget-class>
 
     <glade-widget-class name="GtkButtonBox" _title="Button Box">
       <properties>
@@ -1516,27 +1545,38 @@ embedded in another object</_tooltip>
       </properties>
     </glade-widget-class>
     
-    <glade-widget-class name="GtkHButtonBox" generic-name="hbuttonbox" _title="Horizontal Button Box"/>
-
-    <glade-widget-class name="GtkVButtonBox" generic-name="vbuttonbox" _title="Vertical Button Box"/>
+    <glade-widget-class name="GtkHButtonBox" generic-name="hbuttonbox" _title="Horizontal Button Box">
+      <properties>
+	<property id="orientation" disabled="True"/>
+      </properties>
+    </glade-widget-class>
+    <glade-widget-class name="GtkVButtonBox" generic-name="vbuttonbox" _title="Vertical Button Box">
+      <properties>
+	<property id="orientation" disabled="True"/>
+      </properties>
+    </glade-widget-class>
 
     <glade-widget-class name="GtkHSeparator" generic-name="hseparator" _title="Horizontal Separator">
-      
+      <properties>
+	<property id="orientation" disabled="True"/>
+      </properties>
       <packing-defaults>
 	<parent-class name="GtkVBox">
 	  <child-property id="expand" default="false"/>
 	</parent-class>
       </packing-defaults>
-
     </glade-widget-class>
-
     <glade-widget-class name="GtkVSeparator" generic-name="vseparator" _title="Vertical Separator">
+      <properties>
+	<property id="orientation" disabled="True"/>
+      </properties>
       <packing-defaults>
 	<parent-class name="GtkHBox">
 	  <child-property id="expand" default="false"/>
 	</parent-class>
       </packing-defaults>
     </glade-widget-class>
+
     <glade-widget-class name="GtkStatusbar" generic-name="statusbar" _title="Status Bar">
       <properties>
 	<property id="size" disabled="True" />
@@ -1740,9 +1780,16 @@ embedded in another object</_tooltip>
       </properties>
     </glade-widget-class>
 
-    <glade-widget-class name="GtkHRuler" generic-name="hruler" _title="Horizontal Ruler"/>
-    
-    <glade-widget-class name="GtkVRuler" generic-name="vruler" _title="Vertical Ruler"/>
+    <glade-widget-class name="GtkHRuler" generic-name="hruler" _title="Horizontal Ruler">
+      <properties>
+	<property id="orientation" disabled="True"/>
+      </properties>
+    </glade-widget-class>
+    <glade-widget-class name="GtkVRuler" generic-name="vruler" _title="Vertical Ruler">
+      <properties>
+	<property id="orientation" disabled="True"/>
+      </properties>
+    </glade-widget-class>
 
     <glade-widget-class name="GtkCombo" generic-name="combo" _title="Combo" deprecated="True" libglade-only="True">
       <post-create-function>glade_gtk_combo_post_create</post-create-function>
-- 
1.6.4.2

